frontend/src/components/utils/Pagination.tsx

```diff
@@ -61,6 +61,21 @@ export const Pagination = ({
         config
     );
 
+    // When we change the query params there my be a different number of
+    // results. Thus we need when the key change, to reset the page to 1.
+    // However we shouldn't do it before the key is initialize, hence the
+    // prevApiKey !== ""
+    const [prevApiKey, setPrevApiKey] = useState("");
+    useEffect(() => {
+        let apiKeyJSON = JSON.stringify(apiKey);
+        if (apiKeyJSON !== prevApiKey) {
```

Comment:
I know this is a very sensitive topic but you write `Api` but `JSON`. Please choose a standard.
I am not sure if there are similar names in the codebase which could give us a guideline.
FWIW, my inclination goes to `prevApiKey` and `apiKeyJson`.

Reply:
Same